app/src/org/commcare/CommCareApplication.java

```diff
@@ -1051,4 +1056,39 @@ public static CommCareNoficationManager notificationManager() {
         return app.noficationManager;
     }
 
+    public static boolean applyMissedMigration(SQLiteDatabase db, Context c) {

```

Comment:
You probably want this to happen on user input somewhere, not on login. Otherwise this is going to fire over and over and if they fail to send after the first time it'll crash and they'll be locked out again

Reply:
@ctsims its wrapped in a try/catch so it'll fail silently